libraries/rush-lib/src/api/LastInstallFlag.ts

```diff
@@ -40,9 +80,9 @@ export class LastInstallFlag {
    * @param folderPath - the folder that this flag is managing
    * @param state - optional, the state that should be managed or compared
    */
-  public constructor(folderPath: string, state: JsonObject = {}) {
+  public constructor(folderPath: string, state?: Partial<ILastInstallFlagJson>) {
     this.path = path.join(folderPath, this.flagName);
-    this._state = state;
+    this._state = (state || {}) as ILastInstallFlagJson;
```

Comment:
The property without optional sign("?") in "state" should have a default value if the state passed here is wrapped with "Partial".

Reply:
isn't the `|| {}` the default value?